### PR TITLE
Add documentation for BeanRequestConverterFunction

### PR DESCRIPTION
**Pull request by LiYing2010**
related issue: https://github.com/line/armeria/issues/1106

### CODE REVIEW

File: site/src/sphinx/server-annotated-service.rst [hunk 1]
```diff
@@ -434,6 +438,54 @@ converting ``MyRequest``.
             @RequestObject(MySpecialRequestConverter.class) MyRequest myRequest) { ... }
     }
 
+
+Injecting value of parameters and HTTP headers to a Java object
```
Comment (trustin) [inline site/src/sphinx/server-annotated-service.rst]: to -> into

File: site/src/sphinx/server-annotated-service.rst [hunk 2]
```diff
@@ -434,6 +438,54 @@ converting ``MyRequest``.
...
+
+.. code-block:: java
+
+    public class MyAnnotatedServiceClass {
+        @Post("/hello")
+        public HttpResponse hello(@RequestObject MyRequestObject myRequestObject) { ... }
+    }
+
+Beside this, you also need to inject value of path parameters, HTTP parameters, or HTTP headers to the Java
+object, by putting `@Param`_ or `@Header`_ annotations on the following elements of the Java object:
```
Comment (trustin) [inline site/src/sphinx/server-annotated-service.rst]: - on the following -> on any of the following
- elements of the Java object -> elements in the request object

File: site/src/sphinx/server-annotated-service.rst [hunk 3]
```diff
@@ -434,6 +438,54 @@ converting ``MyRequest``.
...
+
+- fields
+- or, constructors with only one parameter
+- or, methods with only one parameter
+- or, parameters of constructors
+- or, parameters of methods
```
Comment (trustin) [inline site/src/sphinx/server-annotated-service.rst]: - Once we said `any of the following ..`, we can drop `or, `.
- Please capitalize the first word of each list item, e.g. fields -> Fields

File: site/src/sphinx/server-annotated-service.rst [hunk 4]
```diff
@@ -434,6 +438,54 @@ converting ``MyRequest``.
...
+
+
+.. code-block:: java
+
+    public class MyRequestObject {
+        @Param("name") // this field will be injected by value of parameter "name"
```
Comment (trustin) [inline site/src/sphinx/server-annotated-service.rst]: - Please start a sentence with a capitalized word, e.g. this -> This
- Please end a sentence with a period.

File: site/src/sphinx/server-annotated-service.rst (hunk 3 above)
Comment (trustin) [reply site/src/sphinx/server-annotated-service.rst]: - parameters of constructors -> Constructor parameters
- parameters of methods -> method parameters

File: site/src/sphinx/server-annotated-service.rst (hunk 1 above)
Comment (LiYing2010) [reply site/src/sphinx/server-annotated-service.rst]: fixed

File: site/src/sphinx/server-annotated-service.rst [hunk 5]
```diff
@@ -434,6 +438,54 @@ converting ``MyRequest``.
...
+`@RequestObject`_ annotation on the parameters which you want to convert.
+
+.. code-block:: java
+
+    public class MyAnnotatedServiceClass {
+        @Post("/hello")
+        public HttpResponse hello(@RequestObject MyRequestObject myRequestObject) { ... }
+    }
+
+Beside this, you also need to inject value of path parameters, HTTP parameters, or HTTP headers to the Java
```
Comment (minwoox) [inline site/src/sphinx/server-annotated-service.rst]: you can also inject values of ?

Comment (trustin) [reply site/src/sphinx/server-annotated-service.rst]: I think he is trying to tell the reader that the reader should also add some annotations to `MyRequestObject`. How about rephrasing this way:

```
Besides the annotated service class, you also need to create ``MyRequestObject`` and add the @Param and @Header annotations to it to inject the path parameters, HTTP parameters or HTTP headers
```

Comment (LiYing2010) [reply site/src/sphinx/server-annotated-service.rst]: Yes, that's what I am trying to say.
I will fix it tomorrow.

Comment (LiYing2010) [reply site/src/sphinx/server-annotated-service.rst]: fixed

File: site/src/sphinx/server-annotated-service.rst [hunk 6]
```diff
@@ -434,6 +438,55 @@ converting ``MyRequest``.
...
+
+.. code-block:: java
+
+    public class MyAnnotatedServiceClass {
+        @Post("/hello")
+        public HttpResponse hello(@RequestObject MyRequestObject myRequestObject) { ... }
+    }
+
+Besides the annotated service class, you also need to create ``MyRequestObject`` and put `@Param`_ or
+`@Header`_ annotations on any of the following elements of it, to inject the path parameters, HTTP parameters
```
Comment (trustin) [inline site/src/sphinx/server-annotated-service.rst]: any of the following elements of it -> any of the following elements?

**Comment by trustin** [comment]
Great job! Will merge once CI passes.

**Comment by trustin** [comment]
Thanks, @LiYing2010 !

**Pull request merged**